Only use errtrace in shells that support it

kampka opened the pull request:
Using the -E / errtrace option from the bash builtins breakes roundup
on shells that do not support it, eg. busyboxs' sh.
The same goes for the bash specific ERR signal which is not POSIX
compliant. Both should only be used if roundup is executed in a shell
that support it.

kampka committed:
Only use errtrace in shells that support it

Using the -E / errtrace option from the bash builtins breakes roundup
on shells that do not support it, eg. busyboxs' sh.
The same goes for the bash specific ERR signal which is not POSIX
compliant. Both should only be used if roundup is executed in a shell
that support it.

bmizerany commented on roundup.sh:
```diff
@@ -258,9 +258,11 @@ do
-                set -E
-                trap 'rc=$?; set +x; set -o | grep "errexit.*on" >/dev/null && exit $rc' ERR
+                # But, only do this if the error handling is activated and only if the shell supports it
+                if set -o | grep "^errtrace" >/dev/null; then
```
I think this needs to check that errtrace is "on". My output for `set -o` shows "errtrace off" so this line as is will pass even though there is no errtrace. Is that okay?

kampka replied:
The point of that line is to check whether or not errtrace is a feature that is actually supported by the shell you are using before turning it on in the next nine. Otherwise, small shell implementations like eg. Busybox will fail here if we set an option that is not supported. Therefore, we do not need to care about the current state of errtrace since we will turn it on anyways if we can.

kampka commented:
Any chance we can revisit this? :)

kampka committed:
Avoid bash syntax in test

The function keyword is bash specific and will break tests on any other shell.
This keyword is also purely optional as mentioned in the bash manual,
so removing it will not break anything or have any impact whatsoever.
